stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -318,6 +318,108 @@ private class TraceContext {
     internalConsistencyCheck(returnValueIdx == outputReturnValueCount)
     return traceGraphOutputs
   }
+
+  /// Bind data to the inputs of the graph and return the specialized graph.
+  func specializeTFFunction<Data: TensorGroup>(with data: Data) -> String {
+    // TODO(bgogul): When we have additional inputs, we will have do a bit more.
+    // We will simply cause such cases to fail temporarily.
+    internalConsistencyCheck(additionalInputTensorCount == 0)
+
+    let specializedGraph = TF_NewGraph()!
+
+    TF_GraphCopyFunction(
+      specializedGraph, traceGraphFn, /*gradient*/ nil, status)
+    checkOk(status)
+
+    let tracedFunctionName = TF_FunctionName(traceGraphFn)
+    internalConsistencyCheck(tracedFunctionName != nil)
+    let tracedOpDesc = TF_NewOperation(
+      specializedGraph, tracedFunctionName, "tracedFn")
+
+    // Create and append the inputs to the graph function.
+    let traceeInputs = symbolicInputs.dropLast(
+      Int(data._tensorHandleCount + additionalInputTensorCount))
+    var inputs: [TF_Output] = []
+    for (i, traceeInput) in traceeInputs.enumerated() {
+      let desc = TF_NewOperation(specializedGraph, "Placeholder", "input_\(i)")
+      TF_SetAttrType(desc, "dtype", TF_OperationOutputType(traceeInput))
+      let result = TF_FinishOperation(desc, status)
+      checkOk(status)
+      let input = TF_Output(oper: result, index: 0)
+      TF_AddInput(tracedOpDesc, input)
+      inputs.append(input)
+    }
+
+    // Wire the data to the corresponding inputs of the tracedOp.
+    for (i, cTensorHandle) in data.cTensorHandles.enumerated() {
+      let cTensor = TFE_TensorHandleResolve(cTensorHandle, status)
+      checkOk(status)
+      let desc = TF_NewOperation(specializedGraph, "Const", "input_const_\(i)")
+      TF_SetAttrType(desc, "dtype", TFE_TensorHandleDataType(cTensorHandle))
+      TF_SetAttrTensor(desc, "value", cTensor, status)
+      checkOk(status)
+      let result = TF_FinishOperation(desc, status)
+      checkOk(status)
+      TF_AddInput(tracedOpDesc, TF_Output(oper: result, index: 0))
+    }
+
+    // TODO(bgogul): additional input tensors.
+
+    let tracedOp = TF_FinishOperation(tracedOpDesc, status)
+    checkOk(status)
+
+    // Set up outputs
```

Comment:
Minor: other comments end with a period.
```suggestion
    // Set up outputs.
```